internal/envoy/secrets.go

```diff
@@ -301,29 +301,38 @@ func (s *secretManager) unwatch(names []string, node string) error {
 }
 
 // Manage is used for re-fetching expiring TLS certificates and updating them
-func (s *secretManager) Manage(ctx context.Context) {
+func (s *secretManager) Manage(ctx context.Context, forceInterval time.Duration) {
 	s.logger.Trace("running secrets manager")
 
+	lastForce := time.Now()
+
 	for {
 		select {
 		case <-time.After(s.loopTimeout):
-			s.manage(ctx)
+			force := false
+			now := time.Now()
+			if now.Sub(lastForce) >= forceInterval {
+				force = true
+				lastForce = now
+			}
+			s.manage(ctx, force)
 		case <-ctx.Done():
 			// we finished the context, just return
 			return
 		}
```

Comment:
I think you could simplify a bit at the risk of pulling twice rapidly due to the offset between the two being random. I don't imagine that would be an actual problem operationally though

```go
select {
case <-time.After(s.loopTimeout):
    s.manage(ctx, false)
case <-time.After(forceInterval):
    s.manage(ctx, true)
case <-ctx.Done():
    // we finished the context, just return
    return
}
```